### PR TITLE
Improve golangci-lint configurations

### LINKED ISSUE #1860: Improve golangci-lint configuration
I just discovered golangci has more to offer than what we are currently using.
It may make our lane slower, but the payback is a better quality of the code.

We should investigate and add those parameters (I guess it's fine to add more linters and see the outcome, it doesn't have to be perfect but if it improves the current status it's still a win)

### LINKED ISSUE #1867: Bump golangci-lint to latest to include the new Ginkgo lint
The latest golangci-lint includes a lint for all of the ginkgo framework, we should consider bumping
the version to the latest under the function `lint` in `tasks.py`

see github.com/nunnatsa/ginkgolinter for more info

### PR DESCRIPTION
**Pull request by liornoy**
This PR adds a couple of linters from golangci-lint:

  - ginkgolinter
  - dogsled
  - exportloopref
  - gocritic
  - misspell
  - nolintlint
  - stylecheck
  - unconvert
  - unparam
  - whitespace
  - revive
  - unused
  - wastedassign
Each commit shows what each linter fixes.
The commits are split for sake of reviewing, will squash them if required.
More details about each linter and their configurations can be found here: https://golangci-lint.run/usage/linters/ 

fixes #1860 
fixes #1867

### CODE REVIEW

**Comment by fedepaol** [comment]
Since you are working on this, mind extending and taking https://github.com/metallb/metallb/issues/1860 ?

**Comment by mlguerrero12** [comment]
add revive as well please

**Comment by liornoy** [comment]
> add revive as well please

Done. great linter!

**Comment by liornoy** [comment]
Note: those are mostly "low-hanging fruits" linters.
In the future, we might want to bring more restrictive ones like:
lll - limits the line length of the code
goconst - convert any variable that repeats it self more than once into a const.
gomnd - for magic numbers.
funlen - limits the length of functions.

File: tasks.py [hunk 1]
```diff
@@ -691,7 +691,7 @@ def lint(ctx, env="container"):
     convenient to install the golangci-lint binaries on the host. This can be
     achieved by running `inv lint --env host`.
     """
-    version = "1.50.1"
+    version = "1.51.2"
```
Comment (mlguerrero12) [inline tasks.py:694]: could you run 1.52.1? I've seen some additional restrictions in revive with this version

Comment (liornoy) [reply tasks.py:694]: sure

Comment (liornoy) [reply tasks.py]: done. it didn't require me to make any changes.

**Comment by liornoy** [comment]
> can you run "unused" and "wastedassign" please?

Sure, added.

**Comment by fedepaol** [comment]
thanks!

**Pull request merged**